addons/account_check_printing/models/account_payment.py

```diff
@@ -41,7 +41,7 @@ class AccountPayment(models.Model):
 
     @api.constrains('check_number', 'journal_id')
     def _constrains_check_number(self):
-        if not self:
+        if not self.filtered('check_number'):
```

Comment:
some update to increase speed to check only payment with check number.

```python
payments_with_check = self.filtered('check_number')
if not payments_with_check:
     return
try:
     payments_with_check.mapped(lambda p: str(int(p.check_number)))
except:
     raise ValidationError(_('Check numbers can only consist of digits'))
self.flush()
self.env.cr.execute("""
# add this to speed up:
AND other_payment.check_number IS NOT NULL
""", {
    'ids': tuple(payments_with_check.ids),
})
```

Reply:
>  @fla-subteno-it 
> and you don't need AND payment.check_number IS NOT NUL

Why do you think that?
Do you think it is redundant or returns different results or not performance improvement?


`...AND other_payment.check_number IS NOT NULL`
 Planning Time: 3.660 ms
 Execution Time: 1867.153 ms

`...AND other_payment.check_number IS NOT NULL AND payment.check_number IS NOT NULL`
 Planning Time: 4.699 ms
 Execution Time: 0.192 ms

`...AND payment.check_number IS NOT NULL`
 Planning Time: 3.798 ms
 Execution Time: 0.156 ms